utils/data.hpp

```diff
@@ -0,0 +1,143 @@
+const QString POLYGON = "Polygon";
+const QString RECTANGLE = "Rectangle";
+const QString SQUARE = "Square";
+const QString ELLIPSE = "Ellipse";
+const QString CIRCLE = "Circle";
+const QString TEXT = "Text";
+const QStringList TYPE_LIST = {LINE, POLYLINE, POLYGON, RECTANGLE, SQUARE, ELLIPSE, CIRCLE, TEXT };
+
+// The parser expects this character at the end of descriptive titles for data
+const QChar TITLE_INTERRUPTOR = ' ';
+// The parser expects this character to interrupt each piece of data in the line of text that gives the shape's dimensions
+const QChar DIMENSION_INTERRUPTOR = ',';
+
+// Pairs of parallel lists indirectly associate strings with corresponding values
+const QStringList PEN_STYLE_STR = {
+  "NoPen", "SolidLine", "DashLine", "DotLine",
+    "DashDotLine", "DashDotDotLine", "CustomDashLine"
+};
+const QList<Qt::PenStyle> PEN_STYLE_VAL = {
+  Qt::NoPen, Qt::SolidLine, Qt::DashLine, Qt::DotLine,
+    Qt::DashDotLine, Qt::DashDotDotLine, Qt::CustomDashLine
+};
+const QStringList PEN_CAP_STR = {
+    "SquareCap", "FlatCap", "RoundCap"
+};
+const QList<Qt::PenCapStyle> PEN_CAP_VAL = {
+  Qt::SquareCap, Qt::FlatCap, Qt::RoundCap
+};
+const QStringList PEN_JOIN_STR = {
+  "BevelJoin", "MiterJoin", "RoundJoin"
+};
+const QList<Qt::PenJoinStyle> PEN_JOIN_VAL = {
+  Qt::BevelJoin, Qt::MiterJoin, Qt::RoundJoin
+};
+const QStringList BRUSH_STYLE_STR = {
+  "SolidPattern", "Dense1Pattern", "Dense2Pattern",
+    "Dense3Pattern", "Dense4Pattern", "Dense5Pattern",
+    "Dense6Pattern", "Dense7Pattern", "NoBrush",
+    "HorPattern", "VerPattern", "CrossPattern",
+    "BDiagPattern", "FDiagPattern", "DiagCrossPattern",
+    "LinearGradientPattern", "RadialGradientPattern",
+    "ConicalGradientPattern"
+};
+const QList<Qt::BrushStyle> BRUSH_STYLE_VAL = {
+  Qt::SolidPattern, Qt::Dense1Pattern, Qt::Dense2Pattern,
+    Qt::Dense3Pattern, Qt::Dense4Pattern, Qt::Dense5Pattern,
+    Qt::Dense6Pattern, Qt::Dense7Pattern, Qt::NoBrush,
+    Qt::HorPattern, Qt::VerPattern, Qt::CrossPattern,
+    Qt::BDiagPattern, Qt::FDiagPattern, Qt::DiagCrossPattern,
+    Qt::LinearGradientPattern, Qt::RadialGradientPattern,
+    Qt::ConicalGradientPattern
+};
+const QStringList TEXT_ALIGN_STR = {
+    "AlignLeft", "AlignRight", "AlignTop",
+    "AlignBottom", "AlignCenter"
+};
+const QList<Qt::AlignmentFlag> TEXT_ALIGN_VAL = {
+  Qt::AlignLeft, Qt::AlignRight, Qt::AlignTop,
+    Qt::AlignBottom, Qt::AlignCenter
+};
+const QStringList TEXT_STYLE_STR = {
+  "StyleNormal", "StyleItalic", "StyleOblique"
+};
+const QList<QFont::Style> TEXT_STYLE_VAL = {
+  QFont::StyleNormal, QFont::StyleItalic, QFont::StyleOblique
+};
+const QStringList TEXT_WEIGHT_STR = {
+  "Thin", "ExtraLight", "Light", "Normal", "Medium",
+    "DemiBold", "Bold", "ExtraBold", "Black"
+};
+const QList<QFont::Weight> TEXT_WEIGHT_VAL = {
+  QFont::Thin, QFont::ExtraLight, QFont::Light, QFont::Normal, QFont::Medium,
+    QFont::DemiBold, QFont::Bold, QFont::ExtraBold, QFont::Black
+};
+
+class Data
```

Comment:
I think Data is too ambiguous of a class name. What about something like `Document`?

Reply:
How about `class ShapeData`?